config/pitest-suppressions/pitest-xpath-suppressions.xml

```diff
@@ -9,15 +9,6 @@
     <lineContent>return Collections.unmodifiableList(siblings.subList(0, indexAmongSiblings));</lineContent>
   </mutation>
 
-  <mutation unstable="false">
-    <sourceFile>AbstractElementNode.java</sourceFile>
-    <mutatedClass>com.puppycrawl.tools.checkstyle.xpath.AbstractElementNode</mutatedClass>
-    <mutatedMethod>iterateAxis</mutatedMethod>
-    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.RemoveSwitchMutator_8</mutator>
-    <description>RemoveSwitch 8 (case value 8)</description>
-    <lineContent>switch (axisNumber) {</lineContent>
```

Comment:
@romani, @Vyom-Yadav their is no case value 8 in code https://kevin222004.github.io/Report/com.puppycrawl.tools.checkstyle.xpath/AbstractElementNode.java.html#:~:text=1.%20Changed,value%209)%20%E2%86%92%20KILLED

Reply:
Why didn't CI fail then if this suppression is unnecessary?

Reply:
@rnveach I have faced this kind of  issue before at https://github.com/checkstyle/checkstyle/issues/12331#issuecomment-1379531007 

 I am not sure why ci is not failing 
 
 but as per my search case value 8 is for `AxisInfo.NAMESPACE:` which we are not using in this method 
 
 May be related to https://github.com/hcoles/pitest/issues/1137

Reply:
@Vyom-Yadav , please take a look a this, this is looks weird.
it sounds like we do not use `git diff` for validation of new version suppressions.
Does it mean that we might have bunch of stale suppressions ?

Reply:
@Vyom-Yadav , it was added from the beging https://github.com/checkstyle/checkstyle/commit/30f5044df52a47d5444affff0c3b31f76dc824ff

I think it make sense to remove `if: failure()` for "Patch ..." step

Reply:
after execution of pitest on my local, here is diff:
```
✔ ~/java/github/romani/checkstyle [Kevin222004/xp1 L|✚ 1⚑ 1] 
git diff
diff --git a/config/pitest-suppressions/pitest-xpath-suppressions.xml b/config/pitest-suppressions/pitest-xpath-suppressions.xml
index 56c63a5..e1ca5eb 100644
--- a/config/pitest-suppressions/pitest-xpath-suppressions.xml
+++ b/config/pitest-suppressions/pitest-xpath-suppressions.xml
@@ -9,7 +9,7 @@
     <lineContent>return Collections.unmodifiableList(siblings.subList(0, indexAmongSiblings));</lineContent>
   </mutation>
 
-  <mutation unstable="true">
+  <mutation unstable="false">
     <sourceFile>AbstractRootNode.java</sourceFile>
     <mutatedClass>com.puppycrawl.tools.checkstyle.xpath.AbstractRootNode</mutatedClass>
     <mutatedMethod>iterateAxis</mutatedMethod>
@@ -18,7 +18,7 @@
     <lineContent>switch (axisNumber) {</lineContent>
   </mutation>
 
-  <mutation unstable="true">
+  <mutation unstable="false">
     <sourceFile>AbstractRootNode.java</sourceFile>
     <mutatedClass>com.puppycrawl.tools.checkstyle.xpath.AbstractRootNode</mutatedClass>
     <mutatedMethod>iterateAxis</mutatedMethod>
@@ -27,7 +27,7 @@
     <lineContent>switch (axisNumber) {</lineContent>
   </mutation>
 
-  <mutation unstable="true">
+  <mutation unstable="false">
     <sourceFile>AbstractRootNode.java</sourceFile>
     <mutatedClass>com.puppycrawl.tools.checkstyle.xpath.AbstractRootNode</mutatedClass>
     <mutatedMethod>iterateAxis</mutatedMethod>
@@ -36,15 +36,6 @@
     <lineContent>switch (axisNumber) {</lineContent>
   </mutation>
 
-  <mutation unstable="true">
-    <sourceFile>AbstractRootNode.java</sourceFile>
-    <mutatedClass>com.puppycrawl.tools.checkstyle.xpath.AbstractRootNode</mutatedClass>
-    <mutatedMethod>iterateAxis</mutatedMethod>
-    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.RemoveSwitchMutator_8</mutator>
-    <description>RemoveSwitch 8 (case value 8)</description>
-    <lineContent>switch (axisNumber) {</lineContent>
-  </mutation>
-
   <mutation unstable="false">
     <sourceFile>DescendantIterator.java</sourceFile>
     <mutatedClass>com.puppycrawl.tools.checkstyle.xpath.iterators.DescendantIterator</mutatedClass>

```

Reply:
Unstable true/false is probably why the check doesn't run on a success. The generated form of the file doesn't know what is stable or unstable.

Reply:
@romani @rnveach The problem is that the script automatically removes useless suppressions, and we only do git diff on failure.

Instead, we should do git diff every time, and if it is non-empty then fail.

Reply:
I opened https://github.com/checkstyle/checkstyle/pull/13492

Reply:
https://github.com/checkstyle/checkstyle/pull/13492#issuecomment-1659156472 shows my comment isn't outdated. Git diff doesn't work with unstable mutations.